src/GafferSceneUI/LightTool.cpp

```diff
@@ -158,6 +168,50 @@ void setValueOrAddKey( FloatPlug *plug, float time, float value )
 	}
 }
 
+bool inspectionEditable( const GafferSceneUI::Private::Inspector::Result *inspection )
+{
+	if( !inspection->editable() )
+	{
+		return false;
+	}
+
+	EditScope *editScope = inspection->editScope();
+	if(
+		( editScope && inspection->source() && editScope->isAncestorOf( inspection->source() ) ) ||
+		!editScope
+	)
```

Comment:
Took me a while to get my head round this, so let me see if I've got it straight : we're detecting the case where `source()` is the same plug that `acquireEdit()` will return? The `Queries` and `Editing` sections of the `Result` API were originally meant to be fairly distinct so the fact that we're having to mix them up here in such a fiddly way does seem to imply a deficiency in the Inspector API.

> Should the inspectionEditable() code be moved to Inspection in some way, so clients don't have to know to make those extra checks?

Yes, having reviewed the current Inspector behaviour, I think that would be a good call. The Inspector is already responsible for treating read-only plugs as non-editable, and being read-only is arguably a weaker non-editability than the plug having an input. It's relatively trivial for the Inspector to add this check in `inspectHistoryWalk()`, and doing it there will feed naturally into showing the warning that explains why something's not editable.

I think we ended up here because the original use cases for the Inspector API all involved showing a PlugValueWidget for the user to manipulate themselves, so we got away with a loose definition of editability. It might be a slight concern that in the LightEditor we'll no longer be popping up a PlugValueWidget for plugs with inputs, because you could use that popup to resolve the situation (by removing the input). But the same argument would apply to read-only editability, and we're not worrying about that. And I think there's a reasonable argument that resolving non-editability in the LightEditor without a good view of the wider node-graph context might be a bit dangerous. So using the history traceback or a new "Edit Source Node" function to do that might be a better workflow anyway.

Improving the Inspector behaviour now also paves the way for a future `Result::applyEdit( const Data *newValue )` method which would completely abstract away the nuts and bolts of editing.